### ISSUE TITLE
Nice visualizations!

### ISSUE THREAD

**Issue by glenn-jocher** [SeanAvery/yolov5-tensorrt#4]
The grid visualizations could be very useful for development and testing, and if fact YOLOv5 uses a new multiple-grid assignment structure that shows up well in your images.

If you have a concise plotting function for creating these perhaps we could do a PR into yolov5/utils/general.py (where all the plotting functions are located).

**Comment by SeanAvery** [comment]
@glenn-jocher Thanks dude!

Yeah I think it could be useful, I finally was able to understood the model output when I did the grid visualizations.

That sounds great. I still need to clean up the code some more, but will reach out with a PR this week.

**Comment by glenn-jocher** [comment]
@SeanAvery great, sounds good! Introspection is very important to help understand what is going.

**Comment by PonyPC** [comment]
Best works!

**Comment by SeanAvery** [comment]
@glenn-jocher hey, just a little update. getting back to the yolo code

Forked off from your repository and working on implementing a plotting function in [general.py](https://github.com/ultralytics/yolov5/blob/master/utils/general.py#L1092) called `plot_object_probability` 

One consideration, I am sending the `pred` output before [non_max_suppression](https://github.com/ultralytics/yolov5/blob/master/detect.py#L75) is applied in order to maintain the raw output from forward pass. 

```
Example Ops:
1. input (1, 25200, 85)
2. -> [(1, 3, nx0, ny0, 85), (1, 3, nx1, ny1, 85), (1, 3, nx2, ny2, 85)] deconstruct the stacked output
3. -> [(1, nx0, ny0, 1), (1, nx1, nx2, 1), (1, nx2, ny2, 1)] take max object probability of the the grid anchor sizes per grid cell 
4. -> draw transparent rectangles in a threshold or hue range of the probability score
```
I was thinking it would be cool if there was a model "debug" mode where visualizations like the object probability, class grid, raw bounding boxes, and anchor statistic could be processed.

Making sure to use matplot lib to write images to inference/output directory where they can easily be displayed in the notebook.

**Comment by SeanAvery** [comment]
Object probability grid refactored to pytorch and working in the codebase. works on cpu and gpu

[plot_object_probability()](https://github.com/SeanAvery/yolov5/blob/exp/utils/general.py#L1112)

cli argument `--mdebug` to dispatch the visualization

[detect.py](https://github.com/SeanAvery/yolov5/blob/exp/detect.py#L168)

I am still working on removing all hardcoding and simplifying the code more.

**Comment by glenn-jocher** [comment]
@SeanAvery oh that's really cool. Can you show some examples?

Yes, a debug mode would be interesting. If we can keep all of the plots seperated in general.py, or perhaps bring all plotting functions into their own plots.py, so the inline impact is minimal (i.e. in the loss function), then I think this could work well.

**Comment by SeanAvery** [comment]
![20_20_object_prob](https://user-images.githubusercontent.com/12690131/92186027-71478500-ee23-11ea-8edf-40202f85b844.jpg)
![40_40_object_prob](https://user-images.githubusercontent.com/12690131/92186028-7278b200-ee23-11ea-9130-5daedacafbd6.jpg)
![40_40_object_prob](https://user-images.githubusercontent.com/12690131/92186059-8d4b2680-ee23-11ea-836e-709f48f55c43.jpg)

**Comment by henbucuoshanghai** [comment]
/home/aipanda/Pictures/Screenshot from 2021-06-03 19-11-15.png
/home/aipanda/Pictures/Screenshot from 2021-06-03 19-11-24.png

but where is the box of instances?

**Comment by henbucuoshanghai** [comment]
@SeanAvery    only output the   heatmap?   where is the box of instances?
